htdocs/class/cache/file.php

```
@@ -211,9 +211,25 @@ public function read($key)
+            // Try the payload as-is first. The length-repair pass below exists for
+            // magic_quotes-era data, but its pattern is non-greedy: a stored string
+            // containing '";' terminates the match early, so the rewritten length is
+            // too short and unserialize() then fails at an offset. That silently
+            // turned every affected cache entry into a permanent miss.
+            $unserialized = unserialize($data, ['allowed_classes' => false]);
+
+            if (false === $unserialized && 'b:0;' !== $data) {
+                // Genuinely malformed: fall back to the legacy length repair.
+                $repaired = preg_replace_callback(
+                    '!s:(\d+):"(.*?)";!s',
+                    static function ($m) { return 's:' . strlen($m[2]) . ':"' . $m[2] . '";'; },
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $m. Configured minimum length is 3. (undefined)

(ShortVariable)